RecoLocalCalo/HGCalRecAlgos/src/HGCalImagingAlgo.cc

```diff
@@ -172,15 +172,15 @@ math::XYZPoint HGCalImagingAlgo::calculatePosition(std::vector<KDNode> &v){
 } 
 
 double HGCalImagingAlgo::distance(const Hexel &pt1, const Hexel &pt2){
-  const GlobalPoint position1( std::move( rhtools_.getPosition( pt1.detid ) ) );
-  const GlobalPoint position2( std::move( rhtools_.getPosition( pt2.detid ) ) );
-  return sqrt(pow(pt1.x - pt2.x, 2) + pow(pt1.y - pt2.y, 2));
+  const double dx = pt1.x - pt2.x;
+  const double dy = pt1.y - pt2.y;
+  return std::sqrt(dx*dx + dy*dy);
```

Comment:
is distance needed primarily to select or order elements?
If the actual sqrt is needed in a much smaller fraction of cases, better compute distance^2
and the apply sqrt in that smaller fraction of cases.